controllers/imagerepository_controller.go

```diff
@@ -161,6 +220,15 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}
 
+	// we are adding to map only for new provision, not for some partial actions,
+	// so report time only if time was recorded
+	_, timeRecorded := repositoryTimesForMetrics[componentIdForMetrics]
+	if timeRecorded {
+		imageRepositoryProvisionTimeMetric.Observe(time.Since(repositoryTimesForMetrics[componentIdForMetrics]).Seconds())
```

Comment:
This code a bit twisted.
1. The ignored flag can be used in the if condition
2. `timeRecorded` is already here, no need to query map again.

Reply:
ok I will not ignore value and then not query it again